### ISSUE TITLE
Embedded PDB causes VS to jump to Shouldly internals instead of user test code when debugging assertions

### ISSUE THREAD

**Issue by jnm2** [shouldly/shouldly#769]
This is caused by VS not following the "Just My Code" setting and treating any DLL that contains embedded PDB as the user's code.

We had the same problem at NUnit and decided to go the route of the new recommended snupkg symbol package format.

https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/nuget#symbol-packages
https://docs.microsoft.com/en-us/nuget/create-packages/symbol-packages-snupkg
https://devblogs.microsoft.com/nuget/improved-package-debugging-experience-with-the-nuget-org-symbol-server/

**Comment by SimonCropp** [comment]
this seems like a bug in VS. and not something libraries should need to work around

**Comment by jnm2** [comment]
I agree, and yet practically speaking it's so nasty on a regular basis.

**Comment by SimonCropp** [comment]
yep but you should be raising it with the VS team. given many OSS projects are shipping with embedded symbols, shouldly no using embedded symbols will only slightly mitigate the issue

**Comment by jnm2** [comment]
Debugging a test stops you in an inappropriate stack frame whenever an assertion fails. That happens at a rate far beyond any other library I can think of in the course of any kind of debugging (app or tests).

**Comment by SimonCropp** [comment]
all the more reason to raise a bug with the VS team so they can fix it asap

**Comment by jnm2** [comment]
That's for sure. A both-and strategy solves the immediate need, and also helps people who don't stay on (or can't use) the latest VS all the time.

**Comment by SimonCropp** [comment]
2 years later. is this still a bug in VS. FWIW i am not a fan of the added complexity to work around a VS bug

**Comment by SimonCropp** [comment]
closing this and can re-open if it is still occurring and someone re-raises
